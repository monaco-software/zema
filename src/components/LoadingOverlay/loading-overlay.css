.loading-overlay {
  position: relative;
}

.loading-overlay_show .loading-overlay__children {
  opacity: 0.4;
  pointer-events: none;
  user-select: none;
}

.loading-overlay__layer {
  display: flex;
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  align-items: center;
  justify-content: center;
}
