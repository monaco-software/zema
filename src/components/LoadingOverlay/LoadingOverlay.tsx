import './loading-overlay.css';
import b_ from 'b_';
import React, { FC } from 'react';
import { Spinner, SpinnerProps } from '../Spinner/Spinner';

const block = b_.lock('loading-overlay');

interface Props extends SpinnerProps {
  isLoading: boolean;
}

export const LoadingOverlay: FC<Props> = ({
  isLoading,
  children,
  ...restProps
}) => {
  return (
    <div className={block({ show: isLoading })}>
      <div className={block('children')}>{children}</div>
      {isLoading && (
        <div className={block('layer')}>
          <Spinner {...restProps} />
        </div>
      )}
    </div>
  );
};
