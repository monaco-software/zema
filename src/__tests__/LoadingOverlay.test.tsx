import React from 'react';
import renderer from 'react-test-renderer';
import { LoadingOverlay } from '@components/LoadingOverlay/LoadingOverlay';

it('<LoadingOverlay /> should render content', () => {
<<<<<<< HEAD
  const tree = renderer.create(
    <LoadingOverlay isLoading={false}>
      some content
    </LoadingOverlay>
  ).toJSON();
=======
  const tree = renderer
    .create(<LoadingOverlay isLoading={false}>some content</LoadingOverlay>)
    .toJSON();
>>>>>>> dev

  expect(tree).toMatchSnapshot();
});

it('<LoadingOverlay /> should render content with overlay', () => {
<<<<<<< HEAD
  const tree = renderer.create(
    <LoadingOverlay isLoading={true}>
      some content with loading overlay
    </LoadingOverlay>
  ).toJSON();
=======
  const tree = renderer
    .create(
      <LoadingOverlay isLoading={true}>
        some content with loading overlay
      </LoadingOverlay>
    )
    .toJSON();
>>>>>>> dev

  expect(tree).toMatchSnapshot();
});
