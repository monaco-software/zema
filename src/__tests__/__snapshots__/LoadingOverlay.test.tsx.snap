// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingOverlay /> should render content 1`] = `
<div
  className="loading-overlay"
>
  <div
    className="loading-overlay__children"
  >
    some content
  </div>
</div>
`;

exports[`<LoadingOverlay /> should render content with overlay 1`] = `
<div
  className="loading-overlay loading-overlay_show"
>
  <div
    className="loading-overlay__children"
  >
    some content with loading overlay
  </div>
  <div
    className="loading-overlay__layer"
  >
    <svg
      fill="#7D4CDB"
      height={28}
      version="1.1"
      viewBox="0 0 32 32"
      width={28}
    >
      <path
        d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
        opacity=".25"
      />
      <path
        d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
      >
        <animateTransform
          attributeName="transform"
          dur="0.8s"
          from="0 16 16"
          repeatCount="indefinite"
          to="360 16 16"
          type="rotate"
        />
      </path>
    </svg>
  </div>
</div>
`;
